File: nhglib-core/src/io/github/movementspeed/nhglib/runtime/messaging/Message.java
package io.github.movementspeed.nhglib.runtime.messaging;

import io.github.movementspeed.nhglib.utils.data.Bundle;
import io.github.movementspeed.nhglib.utils.data.StringUtils;

/**
 * Created by Fausto Napoli on 01/11/2016.
 * Data structure for a message.
 */
public class Message {
    public int id;
    public Bundle data;

    public Message(String name) {
        this(name, null);
    }

    public Message(String name, Bundle data) {
        id = StringUtils.idFromString(name);
        this.data = data;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Message message = (Message) o;
        return id == message.id;
    }

    @Override
    public int hashCode() {
        int result = id;
        result = 31 * result + (data != null ? data.hashCode() : 0);
        return result;
    }

    public boolean is(String name) {
        int id = StringUtils.idFromString(name);
        return this.id == id;
    }
}

File: nhglib-core/src/io/github/movementspeed/nhglib/core/messaging/Message.kt
package io.github.movementspeed.nhglib.core.messaging

import io.github.movementspeed.nhglib.utils.data.Bundle
import io.github.movementspeed.nhglib.utils.data.StringUtils

/**
 * Created by Fausto Napoli on 01/11/2016.
 * Data structure for a message.
 */
class Message @JvmOverloads constructor(name: String, var data: Bundle? = null) {
    var id: Int = 0

    init {
        id = StringUtils.idFromString(name)
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false

        val message = o as Message?
        return id == message!!.id
    }

    override fun hashCode(): Int {
        var result = id
        result = 31 * result + if (data != null) data!!.hashCode() else 0
        return result
    }

    fun `is`(name: String): Boolean {
        val id = StringUtils.idFromString(name)
        return this.id == id
    }
}
